vpr/src/pack/re_cluster.cpp

```diff
@@ -28,44 +41,137 @@ bool move_atom_to_new_cluster(const AtomBlockId& atom_id,
 
     if (helper_ctx.num_used_type_instances[block_type] == num_instances) {
         VTR_LOG("The utilization of block_type %s is 100%. No space for new clusters\n", block_type->name);
-        VTR_LOG("Atom %d move aborted\n", atom_id);
+        VTR_LOG("Atom %d move aborted\n", molecule->atom_block_ids[molecule->root]);
         return false;
     }
 
-    //remove the atom from its current cluster and check its legality
-    is_removed = remove_atom_from_cluster(atom_id, lb_type_rr_graphs, old_clb, clustering_data, imacro, during_packing);
+    //remove the molecule from its current cluster and check the cluster legality after
+    is_removed = remove_mol_from_cluster(molecule, molecule_size, old_clb, old_router_data);
+
     if (!is_removed) {
-        VTR_LOG("Atom: %zu move failed. Can't remove it from the old cluster\n", atom_id);
+        VTR_LOG("Atom: %zu move failed. Can't remove it from the old cluster\n", molecule->atom_block_ids[molecule->root]);
```

Comment:
Again, I think this should be a VTR_LOGV as it could produce a significant amount of output. It's not really a problem (just a chatty optimizer) so seems more like a VTR_LOGV.